lib/exprotobuf/define_message.ex

```diff
@@ -104,17 +104,20 @@ defmodule Protobuf.DefineMessage do
   end
 
   defp record_fields(fields) do
-    for %Field{name: name, occurrence: occurrence} <- fields do
-      {name, case occurrence do
-        :repeated -> []
-        _ -> nil
-      end}
-    end
-    ++
-    for %OneOfField{name: name} <- fields do
-      {name, nil}
-    end
-
+    fields
+    |> Enum.map(fn(field) ->
+      case field do
+        %Field{name: name, occurrence: :repeated} ->
+          {name, []}
+        %Field{name: name} ->
+          {name, nil}
+        %OneOfField{name: name} ->
+          {name, nil}
+        _ ->
+          nil
+      end
+    end)
+    |> Enum.reject(fn(field) -> is_nil(field) end)
```

Comment:
I know why you added the clause, what I meant is that probably case will never return nil, so Enum.reject is an overkill. I added it because before list comprehension was pattern matched, so there was no possibility that anything else than %Field{} will be processed. I had to remove that matching to add OneOfFields, so I added reject in case something weird was passed into function.